react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { counter: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ counter: this.state.counter + 1 });
+  }
+
+  render() {
+    if (this.state.counter >= 18) {
+      return <button onClick={this.handleClick} className="btn-lvl-7">Hot Button</button>;
+    } else if (this.state.counter >= 15) {
+      return <button onClick={this.handleClick} className="btn-lvl-6">Hot Button</button>;
+    } else if (this.state.counter >= 12) {
+      return <button onClick={this.handleClick} className="btn-lvl-5">Hot Button</button>;
+    } else if (this.state.counter >= 9) {
+      return <button onClick={this.handleClick} className="btn-lvl-4">Hot Button</button>;
+    } else if (this.state.counter >= 6) {
+      return <button onClick={this.handleClick} className="btn-lvl-3">Hot Button</button>;
+    } else if (this.state.counter >= 3) {
+      return <button onClick={this.handleClick} className="btn-lvl-2">Hot Button</button>;
+    } else {
+      return <button onClick={this.handleClick} className="btn-lvl-1">Hot Button</button>;
+    }
```

Comment:
This is a lot of repetition and would be hard to maintain. It would be better to first figure out the className then return a single button.
```JSX
let btnClass = ''

if (this.state.counter >= 18) {
  btnClass = "btn-lvl-7";
} else if (this.state.counter >= 15) {
  btnClass = "btn-lvl-6";
} else if (this.state.counter >= 12) {
  btnClass = "btn-lvl-5";
} else if (this.state.counter >= 9) {
  btnClass = "btn-lvl-4";
} else if (this.state.counter >= 6) {
  btnClass = "btn-lvl-3";
} else if (this.state.counter >= 3) {
  btnClass = "btn-lvl-2";
} else {
  btnClass = "btn-lvl-1";
}

return <button onClick={this.handleClick} className={btnClass}>Hot Button</button>;
```